### DISCUSSION TITLE
Wondering why the dependency to Cocoapods on iOS

### DISCUSSION THREAD

**Discussion by ristkof** [joreilly/PeopleInSpace#114] [Q&A]
When looking at the way the project is setup for iOS, I am wondering why the approach was taken to compile the kotlin common code into a cocoapod module and use that for linking?

Is it just that it was convenient for this particular project, or is it a recommended approach for some specific reason?

**Comment by om-ha** [comment]
Very valid question, I hope others would shed some light here.

From my end, I found these great resources:

1- [Different Approaches in Consuming KMM Modules in iOS](https://medium.com/wantedly-engineering/different-approaches-in-consuming-kmm-modules-in-ios-7957c722b114)
- CocoaPods Gradle plugin and git submodule
- Generate and distribute universal (fat) frameworks
- Using transitive dependencies with Gradle

2- This one is by **John O'Reilly**, the maintainer of this repo! [Using Swift Packages in a Kotlin Multiplatform project](https://johnoreilly.dev/posts/kotlinmultiplatform-swift-package/)
- Swift Packages

**Comment by joreilly** [comment]
When this project was created 2+ years ago this was I think the recommended approach.  Options have increased since then but I believe this is still valid option (one of the 2 you get prompted to use for example when using KMM plugin).  I've used other approaches on other samples here so of course a choice for folks which one to use themselves.
